Reveal design guidelines (#1405)
hashicorp/design-system: website/docs/components/reveal/partials/guidelines/guidelines.md

### When not to use

- To hide information critical to the user’s task.
- To contain complex or advanced content.

## Toggle text

- Ensure that the text clearly describes the content that will be revealed when the toggle is triggered.
- Keep the text short, clear, and concise. Avoid long sentences.
- Use action-oriented language when possible to convey that there is additional content to be shown. e.g. Show more options.

!!! Dont

Don’t use vague or unrelated text to describe the content.

![Reveal example with a form](/assets/components/reveal/toggle-text-dont.png =591x*)
!!!

!!! Do

Use meaningful text that relates to the content inside.

![Reveal example with a form](/assets/components/reveal/toggle-text-do.png =591x*)
!!!

## Content type

The Reveal component should be used for hiding and showing simple content. For complex content and advanced functionality, consider using an Accordion (coming soon).

!!! Dont

![Reveal example with complex content](/assets/components/reveal/content-type-dont.png =725x*)
!!!

!!! Do

![Reveal example with simple content](/assets/components/reveal/content-type-do.png =604x*)
!!!

> ## Spacing

Reviewer: The only feedback I have here is that it seems like this spacing is for a use-case where the reveal is coupled with a specific field; e.g. giving more context/details about the field directly above it. Would the spacing differ if the content within the reveal was related to the entire section of fields?

Author: @jorytindall I redid this part to improve the section and cover more details. Hopefully, it's clearer and more descriptive now. I also updated the images to be more generic/agnostic.

One thing to note:
- I found it hard to give a recommendation on the margin-bottom because it depends on what content is below the component. There could be an element that is part of the same group, a separate section. The Reveal could be at the end of a form, followed by button groups. It could be inside a card with nothing else below, etc...

### When to use

- To hide additional information until it is relevant to users.

### When not to use

- To hide information critical to the user's task.
- To contain complex or advanced content.

## Toggle text

- Ensure that the text clearly describes the content that will be revealed when the toggle is triggered.
- Keep the text short, clear, and concise. Avoid long sentences.

> - Ensure that the text clearly describes the content that will be revealed when the toggle is triggered.

Reviewer 2: ```suggestion
```

Reviewer 2: duplicate text

> - Keep the text short, clear, and concise. Avoid long phrases.

Reviewer 2: ```suggestion
```

Reviewer 2: duplicate text

> Use action-oriented language when possible to convey that there is additional content to be shown. e.g. Show more options.

Reviewer 2: ```suggestion
Use action-oriented language when possible to convey that there is additional content to be shown. e.g. “Show more options”.
```

!!! Dont

> Don't use vague or unrelated text to describe the content.

Reviewer 2: ```suggestion
Don’t use vague or unrelated text to describe the content.
```

![Reveal example with a form](/assets/components/reveal/toggle-text-dont.png)
!!!

!!! Do

Use meaningful text that relates to the content inside.

> ![Reveal example with a form](/assets/components/reveal/toggle-text-do.png)

Reviewer 2: We should try to add the widths to images when we can to ensure they display at the proper dimensions. For retina size (2x) images, the width should be half the actual image pixel width.

(I also recommend cropping excess whitespace surrounding images whenever practical.)

Reviewer 2: The intrinsic size of this image when I inspect it in a browser is 1600 x 660 so we should set a width of "800" (1600 / 2)

`![Reveal example with a form](/assets/components/reveal/toggle-text-do.png =800x*)`

Author: Thanks for this! Until now, I used to have the same size for all images. I just trimmed the whitespace and added their width.
